src/query/storage/m3/storage.go

```diff
@@ -206,31 +195,31 @@ func (s *m3storage) FetchCompressed(
 		result.Metadata.Resolutions = resolutions
 	}
 
-	return result, accumulator.Close, nil
+	return result, cleanup, nil
 }
 
 // fetches compressed series, returning a MultiFetchResult accumulator
 func (s *m3storage) fetchCompressed(
 	ctx context.Context,
 	query *storage.FetchQuery,
 	options *storage.FetchOptions,
-) (consolidators.MultiFetchResult, error) {
+) (consolidators.SeriesFetchResult, []storagemetadata.Attributes, Cleanup, error) {
```

Comment:
nit: add `consolidators.SeriesFetchResult, []storagemetadata.Attributes, Cleanup` to a struct, this is pretty long